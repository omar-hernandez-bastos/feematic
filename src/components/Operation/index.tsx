import React from 'react'

interface Props {
  text: string
}

const Operation = ({ text }: Props) => {
  const regex = /[0-9]/g
  const textArr = [...text]
  return (
    <div className="text-right text-gray-300 font-light text-3xl">
      {textArr.map((char, index) =>
        new RegExp(regex).test(char) ? (
          char
        ) : (
          <span key={index} className="text-cyan-500 font-normal">
            {char}
          </span>
        )
      )}
    </div>
  )
}

export default Operation
