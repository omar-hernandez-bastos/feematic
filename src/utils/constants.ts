export enum ActionsPad {
  CLEAR = 'ðŸš«',
  DECIMAL = '.'
}
export const pads = [
  { text: 'ðŸ‡¨ðŸ‡´', className: '', color: 'cyan-200' },
  { text: 'ðŸ‡ºðŸ‡¸', className: '', color: 'gray-100' },
  { text: 'ðŸ‡»ðŸ‡ª', className: 'row-span-2' },
  { text: '7', className: '' },
  { text: '8', className: '' },
  { text: '9', className: '' },
  { text: '4', className: '' },
  { text: '5', className: '' },
  { text: '6', className: '' },
  { text: '1', className: '' },
  { text: '2', className: '' },
  { text: '3', className: '' },
  { text: ActionsPad.CLEAR, className: 'row-span-2' },
  { text: '0', className: 'col-span-2' },
  { text: ActionsPad.DECIMAL, className: '' }
]

export type CURRENCIES = 'ðŸ‡¨ðŸ‡´' | 'ðŸ‡ºðŸ‡¸' | 'ðŸ‡»ðŸ‡ª'
const currencies = {
  COP: 'ðŸ‡¨ðŸ‡´',
  USD: 'ðŸ‡ºðŸ‡¸',
  VES: 'ðŸ‡»ðŸ‡ª'
}
export const CURRENCIES_ARRAY = [currencies.USD, currencies.COP, currencies.VES]
export { currencies }
