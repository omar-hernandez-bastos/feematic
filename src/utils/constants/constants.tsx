export enum ActionsPad {
  CLEAR = 'AC',
  C = 'C',
  DECIMAL = '.',
  CREDIT_CARD = 'ðŸ’³'
}
export const pads = [
  { id: 1, text: 'ðŸ‡¨ðŸ‡´', className: 'col-span-2', color: 'cyan-200' },
  { id: 2, text: 'ðŸ‡ºðŸ‡¸', className: 'col-span-2', color: 'gray-100' },
  { id: 4, text: '7', className: '' },
  { id: 5, text: '8', className: '' },
  { id: 6, text: '9', className: '' },
  { id: 3, text: 'ðŸ‡»ðŸ‡ª', className: '' },
  { id: 7, text: '4', className: '' },
  { id: 8, text: '5', className: '' },
  { id: 9, text: '6', className: '' },
  { id: 3, text: 'ðŸ’³', className: '' },
  { id: 10, text: '1', className: '' },
  { id: 11, text: '2', className: '' },
  { id: 12, text: '3', className: '' },
  { id: 13, text: ActionsPad.C, className: '' },
  { id: 14, text: '0', className: 'col-span-2' },
  { id: 15, text: ActionsPad.DECIMAL, className: '' },
  { id: 16, text: ActionsPad.CLEAR, className: '' }
]

export type CURRENCIES = 'ðŸ‡¨ðŸ‡´' | 'ðŸ‡ºðŸ‡¸' | 'ðŸ‡»ðŸ‡ª'
const currencies = {
  COP: 'ðŸ‡¨ðŸ‡´',
  USD: 'ðŸ‡ºðŸ‡¸',
  VES: 'ðŸ‡»ðŸ‡ª'
}
export const CURRENCIES_ARRAY = [currencies.USD, currencies.COP, currencies.VES]
export { currencies }

export const COP_DEFAULT = 3800
export const VES_DEFAULT = 4.5
