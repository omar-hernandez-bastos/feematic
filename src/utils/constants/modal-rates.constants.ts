const CUSTOM_VES = 'customVes'
const CUSTOM_COP = 'customCop'
const CUSTOM_RATES = 'customRates'
export { CUSTOM_VES, CUSTOM_COP, CUSTOM_RATES }
