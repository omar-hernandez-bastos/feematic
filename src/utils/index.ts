export function classNames(...classes: unknown[]): string {
  return classes.filter(Boolean).join(' ')
}

export function formatMoney(value: number): string {
  const formatter = new Intl.NumberFormat('es-CO', {
    minimumFractionDigits: 2,
    maximumFractionDigits: 2
  })
  return formatter.format(value)
}
