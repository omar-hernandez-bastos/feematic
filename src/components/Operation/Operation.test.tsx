import { render } from '@testing-library/react'
import { currencies } from 'utils/constants'
import Operation from '.'

describe('Operation Test', () => {
  it('Operation Component should render correctly', () => {
    const OperationProps = {
      amount: {
        value: '1',
        currency: currencies.USD
      }
    }
    const { container } = render(<Operation {...OperationProps} />)
    expect(container).toMatchSnapshot()
  })
})
