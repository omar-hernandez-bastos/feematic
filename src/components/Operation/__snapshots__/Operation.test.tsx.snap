// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Test Operation Component should render correctly 1`] = `
<div>
  <div
    class="text-5xl font-bold text-right text-white"
  >
    1,00
    ðŸ‡ºðŸ‡¸
  </div>
</div>
`;
